Libs/Widgets/ctkLanguageComboBox.cpp

```diff
@@ -162,23 +178,31 @@ void ctkLanguageComboBoxPrivate::updateLanguageItems()
   }
 }
 
+// ----------------------------------------------------------------------------
+QString ctkLanguageComboBoxPrivate::normalizeLocaleCode(const QString& localeCode)
+{
+  QLocale normalized(localeCode);
+  return normalized.name() == "C" ? QString() : normalized.name();
+}
+
 // ----------------------------------------------------------------------------
 bool ctkLanguageComboBoxPrivate::languageItem(const QString& localeCode,
                                               QIcon& icon,
                                               QString& text,
                                               QVariant& data,
                                               bool showCountry)
 {
-  QLocale locale(localeCode);
-  if (localeCode.isEmpty() ||
-      locale.name() == "C")
+
+  if (this->normalizeLocaleCode(localeCode).isEmpty())
     {
     icon = QIcon();
     text = QString();
     data = QVariant();
     return false;
     }
 
+  QLocale locale(localeCode);
```

Comment:
Shouldn't we use the string returned by `this->normalizeLocaleCode(localeCode)` here?

Has the behavior different now for empty or "C" `localeCode` inputs changed? Was it intentional?

Reply:
Behavior is the same has before. Logic was simplified intentionally since:

```
>>> import qt
>>> qt.QLocale("").name() == "C"
True
```

We could have `normalizeLocaleCode` return "C" if the provided local code is invalid, and update this logic to be like ...


```
  if (this->normalizeLocaleCode(localeCode) == "C")
    {
    icon = QIcon();
    text = QString();
    data = QVariant();
    return false;
    }
```

Since `normalizeLocaleCode` is part of the private implementation ... I don't it makes a difference.

Reply:
It is just odd that we call `this->normalizeLocaleCode(localeCode)` but actually don't actually use the returned string (only in an if statement).

I would expect to see something like this:

```
QString normalizedLocaleCode = this->normalizeLocaleCode(localeCode);
if (normalizedLocaleCode.empty())
{
  ...
}
...
QLocale locale(normalizedLocaleCode);
...
```

Reply:
Agreed, the initial intent was to fix the test and have the default vs current language work as expected while minimizing the changes. 

Before, the language combox was not displaying anything after doing the following:

```
w = ctk.ctkLanguageComboBox()
w.defaultLanguage = "en"
w.show()
```

and was leading to the issue now fixed in https://github.com/Slicer/SlicerLanguagePacks/pull/57

Reply:
The assumption was that the languages were set internally by selection and always assumed to be valid.